linalg/slepc.cpp

```diff
@@ -62,6 +62,8 @@ SlepcEigenSolver::SlepcEigenSolver(MPI_Comm comm, const std::string &prefix)
 
 SlepcEigenSolver::~SlepcEigenSolver()
 {
+   if (VR) { delete VR; }
```

Comment:
no need to use `if`. just call `delete`

Reply:
Removed `if` and set pointers to `NULL` after deleting.

Reply:
Setting member pointers to `NULL` in a destructor is unnecessary. Typically, it makes sense in other methods when the object itself is not destroyed.